cluster/kubernetes/sync.go

```diff
@@ -145,7 +150,11 @@ func (c *Cluster) collectGarbage(
 		switch {
 		case !ok: // was not recorded as having been staged for application
 			c.logger.Log("info", "cluster resource not in resources to be synced; deleting", "resource", resourceID)
-			orphanedResources.stage("delete", res.ResourceID(), "<cluster>", res.IdentifyingBytes())
+			if dryRun {
```

Comment:
This might be a bit neater in logs if you just alter the message output above, rather than printing another message -- which might get interleaved with something unrelated. Just adding a `dry-run=true` to the logger (once, at the top of the procedure) would be enough I reckon.

Reply:
That would also serve the purpose, I will do that :)

Reply:
Does this also apply yo line 135?

Reply:
What I usually do in Java, especially in a multithreaded env, is to prefix messages with a context, if needed.

In this case I would do something like:

```
switch {
case !ok && !dryRun:
	c.logger.Log("info", "Garbage-Collection: dry-run, no deletion; cluster resource not in resources to be synced", "resource", resourceID)
	orphanedResources.stage("delete", res.ResourceID(), "<cluster>", res.IdentifyingBytes())
case !ok && dryRun: // was not recorded as having been staged for application
	c.logger.Log("info", "Garbage-Collection: deleting resources; cluster resource not in resources to be synced", "resource", resourceID)
case actual != expected:
	c.logger.Log("warning", "Garbage-Collection: resource to be synced has not been updated; skipping", "resource", resourceID)
	continue
default:
	// The checksum is the same, indicating that it was
	// applied earlier. Leave it alone.
}
```

it's a bit tedious to do this but the benefits are much more clear logs in the end.